### DISCUSSION TITLE
CodeForces: Circuit

### DISCUSSION THREAD

**Discussion by AnymousScripter** [sonsuzus/sonsuzus#513] [Soru Cevap]
Soru:

A. **Zaman Sınırı**: Test başına 1 saniye  
**Bellek Sınırı**: 256 MB  

Alice, n ışık ve 2n anahtardan oluşan bir devre yaptı. Her bileşenin (ışık veya anahtar) iki durumu vardır: açık veya kapalı. Işıklar ve anahtarlar şu şekilde düzenlenmiştir:
- Her ışık tam olarak iki anahtara bağlıdır.
- Her anahtar tam olarak bir ışığa bağlıdır.
- Hangi anahtarın hangi ışığa bağlı olduğu bilinmemektedir.
- Tüm anahtarlar kapalı olduğunda, tüm ışıklar da kapalıdır.
- Bir anahtar açıldığında (açık veya kapalı duruma getirilirse), ona bağlı olan ışığın durumu da değişir.

Alice, devreyi sadece 2n anahtarın durumları ile birlikte kız kardeşi Iris'e getirir ve ona şu bilmecesini verir: "En az ve en fazla kaç ışık açılabilir?"

Küçük kız kardeşinin hareketlerini çok iyi bildiğinden Iris, Alice'e doğru cevabı vermek için bir saniyeden fazla vakit harcamaz. Peki, siz de aynı şeyi yapabilir misiniz?

### Girdi

Her test, birden fazla test vakasından oluşur. İlk satırda tek bir t tam sayısı bulunur (1 ≤ t ≤ 500) — test vakası sayısı. Test vakalarının açıklamaları takip eder.

Her test vakasının ilk satırında tek bir tamsayı n (1 ≤ n ≤ 50) bulunur — devredeki ışık sayısı.

Her test vakasının ikinci satırında 2n tamsayısı a1, a2, …, a2n (0 ≤ ai ≤ 1) bulunur — devredeki anahtarların durumları. ai = 0, i'nci anahtarın kapalı olduğunu, ai = 1, i'nci anahtarın açık olduğunu belirtir.

### Çıktı

Her test vakası için, sırasıyla açılabilecek minimum ve maksimum ışık sayısını belirten iki tam sayı çıktısı verin.

### Örnek

**Girdi:**
```
5
10
0 1 0 1 1 0 1 0 1 0
1
1 1
3
0 1 0 1 0 1
1
1 1 0 0
3
0 1 1 0 0 1
```

**Çıktı:**
```
0 1
1 1
0 2
1 1
2 3
```

### Açıklamalar

- **İlk test vakası**: Devrede sadece bir ışık var ve hiç anahtar açık değil, bu nedenle ışık kesinlikle kapalıdır.
- **İkinci test vakası**: Devrede sadece bir ışık var, fakat ona bağlı olan bir anahtar açık, bu nedenle ışık açıktır.
- **Üçüncü test vakası**: Devrede sadece bir ışık var ve her iki anahtar da açık, bu nedenle ışık iki kez değiştirilip, başlangıçtaki kapalı durumuna geri döner.
- **Dördüncü test vakası**: Hiç ışık açılmaması için anahtarlar şu şekilde düzenlenebilir:
    - Anahtar 1 ve anahtar 4, ışık 1'e bağlıdır. İkisi de kapalı olduğu için ışık 1 de kapalıdır.
    - Anahtar 2 ve anahtar 6, ışık 2'ye bağlıdır. İkisi de kapalı olduğu için ışık 2 de kapalıdır.
    - Anahtar 3 ve anahtar 5, ışık 3'e bağlıdır. İkisi de açık olduğu için ışık 3, başlangıçtaki kapalı durumundan iki kez değiştirilip kapalı kalır.
  
  2 ışığın açık olması için anahtarlar şu şekilde düzenlenebilir:
    - Anahtar 1 ve anahtar 2, ışık 1'e bağlıdır. İkisi de kapalı olduğundan ışık 1 de kapalı kalır.
    - Anahtar 3 ve anahtar 4, ışık 2'ye bağlıdır. Anahtar 3 açık, anahtar 4 kapalı olduğundan ışık 2 bir kez değişir ve açık olur.
    - Anahtar 5 ve anahtar 6, ışık 3'e bağlıdır. Anahtar 5 açık, anahtar 6 kapalı olduğundan ışık 3 bir kez değişir ve açık olur. 

- **Beşinci test vakası**: Bu durumda 2 ışık açılabilir ve maksimumda 3 ışık açılabilir.

---

Çözüm:

```python
def cozul():
    t = int(input())
    for _ in range(t):
        n = int(input())
        anahtarlar = input()
        
        acik_sayi = anahtarlar.count('1')
        maksimum_acik = acik_sayi // 2
        minimum_acik = 1 if acik_sayi % 2 == 1 else 0
        
        print(minimum_acik, maksimum_acik)

cozul()
```

**Comment by sonsuzus** [comment]
Çözümü ayrı bir cevap olarak girmelisin. Ayrıca code bloğuna almayı da unutma.

**Comment by sonsuzus** [comment]
Ayrıca bu soru daha önce yapıldı. Başka soru bulmalısın.
